src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -264,6 +271,52 @@ private void RefreshCondensedDisplay()
             smallContentView.Text = cachedSmallContent; // Update displayed text.
         }
 
+        private void RefreshExpandedDisplayAsync()
+        {
+            var scheduler = nodeViewModel.DynamoViewModel.Model.Scheduler;
+
+            // The preview control will not have its content refreshed unless 
+            // the content is null. In order to perform real refresh, new data 
+            // source needs to be rebound to the preview control by calling 
+            // BindToDataSource method.
+            // 
+            if (this.cachedLargeContent != null)
+                return;
+
+            if (largeContentGrid.Children.Count == 0)
+            {
+                var tree = new WatchTree();
+                tree.DataContext = new WatchViewModel(nodeViewModel.DynamoViewModel.VisualizationManager);
+                largeContentGrid.Children.Add(tree);
+            }
+
+            var watchTree = largeContentGrid.Children[0] as WatchTree;
+            var rootDataContext = watchTree.DataContext as WatchViewModel;
+
+            var task0 = new DelegateBasedAsyncTask(scheduler, () =>
+            {
+                cachedLargeContent = nodeViewModel.DynamoViewModel.WatchHandler.GenerateWatchViewModelForData(
+                    mirrorData, null, string.Empty, false);
```

Comment:
I tried to ignore this but couldn't, `cachedLargeContent` is being set in the context of `SchedulerThread`, but it is also being referred to by other parts of the class (which run on UI thread). I think the safest way to deal with this is to have a local version of `var largeContent`, then use it later on to update `cachedLargeContent` during `Dispatcher.BeginInvoke`. What are your thoughts on this?